Feature/gcg 14

papnoemi199999 opened the pull request:
add: followers

papnoemi199999 committed:
GCG-14: As a user, I want an Add Friend button so that I can send requests.

add: follow/unfollow button

papnoemi199999 committed:
GCG-14: As a user, I want an Add Friend button so that I can send requests.

add: display followers and following counts on profile

papnoemi199999 committed:
GCG-14: As a user, I want an Add Friend button so that I can send requests.

fic: correct href
add: self follow/unfollow rule

krisztina-tropee commented on app/Http/Controllers/FollowerController.php:
```diff
@@ -0,0 +1,20 @@
+    public function store(User $user)
+    {
+        Gate::authorize('follow', $user);
+        auth()->user()->followings()->syncWithoutDetaching([$user->id]);
```
Please return something at the end of controller method. A redirect()->back() if you don't want to do anything specific

krisztina-tropee commented on app/Http/Controllers/FollowerController.php:
```diff
@@ -0,0 +1,20 @@
+    public function destroy(User $user)
+    {
+        Gate::authorize('unfollow', $user);
+        auth()->user()->followings()->detach($user->id);
```
same, return is missing

krisztina-tropee commented on resources/js/pages/Profile.vue:
```diff
@@ -176,6 +180,13 @@ const switchTab = (tab: string) => {
     }
 };
+function follow() {
+    router.post(route('users.follow', props.user.id), {}, { preserveScroll: true });
```
add an isFollowing ref, which becomes true while api is being called, then in the onFinish callback you put it back to false. Put this ref to the button's :loading prop. Same for the unfollow function. If not, I can click on the button quickly and call the backend many times

papnoemi199999 committed:
GCG-14: As a user, I want an Add Friend button so that I can send requests.

fix: add redirect back
fix: prevent multiple follow/unfollow requests with single busy state
